tests/rptest/tests/metrics_reporter_test.py

```diff
@@ -35,21 +34,16 @@ def __init__(self, test_ctx):
                 "metrics_reporter_url": f"{self.http.url}/metrics",
             })
 
-    """
-    Validates key availability properties of the system using a single
-    partition.
-    """
+    def setUp(self):
+        # Start HTTP server before redpanda
+        self.http.start()
+        self.redpanda.start()
```

Comment:
any particular reason for overriding `setUp` here rather than removing the extra `self.redpanda.start` from the test cases? the one thing that comes to mind is that calling `super().setUp()` here for the redpanda case might be useful because `RedpandaTest.setUp` is also responsible now for initial topic creation (though this test doesn't use that feature).

Reply:
The motivation is to get that http.start() in there before redpanda.start(), so that redpanda doesn't fail to send its first few metric reports.

I'm kind of inclined to stick with not calling the superclass setUp, because it feels more robust to have a test say "I'm going to do everything myself" than for the test to fiddle with the setup path but then also assume that the parent's setup method is still safe to call after that.  In this case it's totally equivalent of course.  I've also pondered having a "don't autostart" argument to RedpandaTest.__init__, because almost always when overriding setUp it's for a test to just skip the redpanda start.

Reply:
makes sense